### PR TITLE
Fix potential problem for training

### PR DESCRIPTION
**Pull request by Hexiang-Hu**
Hi, 

It seems like that there is a potential bug in function `check_accuracy`.

This function is called periodically during training to monitor train/val accuracy of VQA models. And at each time, `set_mode('eval', ...)` is called to set model into ***validate*** mode. (Change behaviors for layers such as dropout, batchnorm, etc.). 

However, in its original version, it seems to forget setting model back to ***train*** mode ( by call `set_mode('train', ...)` ) after evaluation, which might cause problems for learning models that have batch-norm or dropout.

Best,

### CODE REVIEW

**Comment by jcjohnson** [comment]
Looks good to me - thanks for the fix!
